src/test/java/org/opensearch/flowframework/workflow/WorkflowProcessSorterTests.java

```diff
@@ -199,4 +202,31 @@ public void testExceptions() throws IOException {
         ex = assertThrows(IllegalArgumentException.class, () -> parse(workflow(List.of(node("A"), node("B")), List.of(edge("A", "C")))));
         assertEquals("Edge destination C does not correspond to a node.", ex.getMessage());
     }
+
+    public void testValidateGraph() {
+
+        // Create Register Model workflow node with missing connector_id field
+        WorkflowNode registerModel = new WorkflowNode(
+            "workflow_step_1",
+            RegisterModelStep.NAME,
+            Map.of(),
+            Map.ofEntries(Map.entry("name", "name"), Map.entry("function_name", "remote"), Map.entry("description", "description"))
+        );
+        WorkflowNode deployModel = new WorkflowNode(
+            "workflow_step_2",
+            RegisterModelStep.NAME,
+            Map.ofEntries(Map.entry("workflow_step_1", "model_id")),
+            Map.of()
+        );
+        WorkflowEdge edge = new WorkflowEdge(registerModel.id(), deployModel.id());
+        Workflow workflow = new Workflow(Map.of(), List.of(registerModel, deployModel), List.of(edge));
+
+        List<ProcessNode> sortedProcessNodes = workflowProcessSorter.sortProcessNodes(workflow);
+        IllegalArgumentException ex = expectThrows(
+            IllegalArgumentException.class,
+            () -> workflowProcessSorter.validateGraph(sortedProcessNodes)
+        );
+        assertEquals("Invalid graph, missing the following required inputs : [connector_id]", ex.getMessage());
+
+    }
```

Comment:
This test the failure case, can we have another test for a successfull case?

Reply:
sure